src/riscv_private.h

```diff
@@ -126,6 +129,18 @@ struct riscv_internal {
     riscv_word_t X[N_RV_REGS];
     riscv_word_t PC;
 
+#if RV32_HAS(JIT) && RV32_HAS(SYSTEM)
+    /*
+     * Aarch encoder only accepts 9 bits signed offset. Do not put this
+     * structure to the bottom.
+     */
+    struct {
+        uint32_t is_mmio; /* whether is MMIO or not */
+        uint32_t type;    /* 0: read, 1: write */
```

Comment:
Would you use bit-fields instead?

Reply:
To simplify the emission of load/store instructions of both `x86-64` and `Aarch64` in `src/rv32_jit.c`, I would like to keep them as unsigned integer.